### PR TITLE
Gate Foodcritic usage behind a feature flag

### PR DESCRIPTION
**Pull request by tyler-ball**
## Description
🚧 For whenever we deprecate Foodcritic.

## Related Issue
Depends on https://github.com/chef/chef-workstation-app/pull/192
Depends on https://github.com/chef/go-libs/pull/31

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have run the pre-merge tests locally and they pass.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

Aha! Link: https://chef.aha.io/features/SH-1844

### CODE REVIEW

File: components/foodcritic-wrapper/go.sum [hunk 1]
```diff
@@ -0,0 +1,128 @@
+cloud.google.com/go v0.26.0/go.mod h1:aQUYkXzVsufM+DwF1aE+0xfcU+56JwCaLick0ClmMTw=
```
Comment (tas50) [inline components/foodcritic-wrapper/go.sum:1]: Is there a way to do this without these large wrappers. This wrapper is going to be bigger than Foodcritic and its deps

**Comment by tas50** [comment]
To circle back to this one I think it's time to start throwing a warning message when running Foodcritic in Workstation so we can start the process of fully removing it.

File: components/foodcritic-wrapper/README.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+Foodcritic has been deprecated and will be removed from Chef Workstation
+in the future. To encourage migration off this tool we have gated it behind
+a feature flag. If the feature flag is disabled, foodcritic will output a
+deprecation message and tell people how to enable the feature flag if they
+want to use it. If the feature flag is enabled, foodcritic works as before.
```
Comment (kagarmoe) [inline components/foodcritic-wrapper/README.md:5]: We might want to give the command.

**Comment by tas50** [comment]
I've removed foodcritic entirely so we can close this one out now

**Pull request closed without merging**